### DISCUSSION TITLE
What is EfficientNet? A quick overview.

### DISCUSSION THREAD

**Discussion by DubLynne** [ultralytics/ultralytics#21875] [Show and tell]
### 📢 New Blog Post: *What is EfficientNet?*  

EfficientNet has quickly become one of the most popular backbone networks in computer vision. But what makes it so powerful, and how can you use it effectively?  

In our latest blog, we break down:  
- How EfficientNet achieves state-of-the-art results with fewer parameters  
- Why compound scaling matters  
- Where it fits into modern deep learning workflows  

👉 Read here: [What is EfficientNet? A Quick Overview](https://www.ultralytics.com/blog/what-is-efficientnet-a-quick-overview)  

[![EfficientNet Overview](https://cdn.prod.website-files.com/680a070c3b99253410dd3df5/68b1d5544eb7c74dbdf2c802_EfficientNET_Thumbnail.webp)](https://www.ultralytics.com/blog/what-is-efficientnet-a-quick-overview)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @DubLynne, thank you for sharing this update with the Ultralytics community 🚀! This is an automated response to help guide next steps while an Ultralytics engineer reviews and will assist you soon.

We recommend a visit to the [Ultralytics Docs](https://docs.ultralytics.com/) for new users where you can find many [Python usage examples](https://docs.ultralytics.com/usage/python/) and [CLI usage examples](https://docs.ultralytics.com/usage/cli/) and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to the [Ultralytics Discord server](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out the [Ultralytics Discourse forum](https://community.ultralytics.com/). Or dive into threads on our [Ultralytics Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements listed in pyproject.toml](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8 environment](https://www.python.org/) with [PyTorch>=1.8 installed](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI checks](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Erez-Yahalomi** [comment]
Actually I explained and proved by 2018 why increasing the number of filters in deep learning layer-width of the layer, increase the accuracy. but the advisor to my Ph.D linger the publication. See my paper first version.
https://arxiv.org/pdf/2001.09203v1

**Reply by glenn-jocher** [reply]
Thanks for sharing, @Erez-Yahalomi—width scaling can help; in practice we see strongest results when depth, width, and input resolution are scaled together (compound scaling), as outlined in our [EfficientDet vs. YOLO11 comparison](https://docs.ultralytics.com/compare/efficientdet-vs-yolo11/); if you can share a small reproducible benchmark isolating width-only gains (dataset, code, metrics), we’ll review here.

**Comment by Erez-Yahalomi** [comment]
Based on a former knowledge of that increase in the number of transistors a in computer chip improves its computing abilities. I made the analogy that the larger the number of parameters or filters in the neural network the higher is its accuracy. This lead me to my research and  paper: Modular network for high accuracy object detection (first version),  https://arxiv.org/pdf/2001.09203v1   . There I proved that the more filters in a deep learning layer, that are dedicated to classify a class, the more accurate the classification. Which in other words, increasing the number of filters in a layer increases the classification accuracy. I proved it in the paper in two ways:  
1. Experimentally,  I compared the classification accuracy of the same object detection deep learning model on two differently fine-tuned networks, first trained to detect 10 classes and the second  trained to detect two classes. The results showed that for the same class, the classification error rate in the 10 classes object detection network is five times larger than the error rate in the 2 classes object detection network.

2. Mathematically, by Bayes error estimation equations and graphs, I showed that a deep learning layer that dedicates a large number of filters to classify each class/category has a smaller classification error compared to a deep learning layer that dedicates less number of filters to classify each class/ category.

**Reply by glenn-jocher** [reply]
Thanks for the added context, @Erez-Yahalomi; the 10‑class vs 2‑class test changes the task’s intrinsic Bayes error, so it doesn’t isolate width—please fix dataset/classes, hold depth and `imgsz` constant, sweep only `width_multiple` in a YOLO11 YAML, then share mAP/FLOPs/latency so we can review; setup details are in the [YOLO11 docs](https://docs.ultralytics.com/models/yolo11/) and the [Train mode guide](https://docs.ultralytics.com/modes/train/).

**Reply by Erez-Yahalomi** [reply]
What I actually proved in this paper: https://arxiv.org/pdf/2001.09203v1  is the more filters used to classify each class the  more accurate is the classification, so it is obvious that increasing  the number of filters in deep learning layer, which is the same as  increasing the width, increases the accuracy. Since the paper Modular network for high accuracy object detection published on arxiv this conclusion was shown in many models including in many LLM and of course in the famous scaling law.

**Reply by glenn-jocher** [reply]
Appreciate the clarification; to validate width-only effects, please run a controlled YOLO11 sweep with fixed data, `imgsz`, depth, schedule, and seed, and vary only `width_multiple` by copying `yolo11n.yaml` to three files (e.g., `w1.00`, `w1.25`, `w1.50`) and training as below, then share final `mAP50-95`, the “Model summary” params/GFLOPs line, and latency from [Benchmark mode](https://docs.ultralytics.com/modes/benchmark/):

```bash
yolo detect train data=path/to/data.yaml model=w1.00.yaml imgsz=640 epochs=100 seed=0 pretrained=False name=w100
yolo detect train data=path/to/data.yaml model=w1.25.yaml imgsz=640 epochs=100 seed=0 pretrained=False name=w125
yolo detect train data=path/to/data.yaml model=w1.50.yaml imgsz=640 epochs=100 seed=0 pretrained=False name=w150

yolo benchmark model=runs/detect/w100/weights/best.pt imgsz=640
yolo benchmark model=runs/detect/w125/weights/best.pt imgsz=640
yolo benchmark model=runs/detect/w150/weights/best.pt imgsz=640
```
